backend/src/main/service/CachingService.ts

```diff
@@ -0,0 +1,52 @@
+import redis from 'redis';
+import { CachingObject } from '../data/CachingObject';
+
+export class CachingService {
+  // todo: change to env later
+  private client = redis.createClient({
+    // host: 'redis-server',
+    host: 'localhost',
```

Comment:
This is where you would change it between Docker and local Redis server!

Reply:
What do you think about creating `client` in App.ts? And then pass the client as a constructor parameter for Caching Service?

Reply:
Hmm.. That's definitely do-able I think!! What were the benefits you were thinking of with regards to that?

Reply:
Thanks! Yeah so the main benefits I was thinking was like:

1. On application start-up, we can detect errors if our Redis client didn't connect rather than detecting errors with Redis when we instantiate and use `CachingService`
2. We maintain a single instance of the `client`. Because right now each time we instantiate `CachingService` we create a client. I'm not sure if `redis.createClient` takes a while or not.
3. I think it's also a bit easier to find in the future where we are connecting to our cache and database if they're at the same place

Let me know if this makes sense!

Reply:
Oooh yeah those are good points! For #2, I don't believe createClient takes awhile but given the benefits of #1/3 I can move it

Reply:
Sounds good thanks! Yah the reason why I thought 2 was because it takes a while for us to establish database connection for TYpeORM